### ISSUE TITLE
Handle logout when user can use many RP applications

### ISSUE THREAD

**Issue by yurem** [GluuFederation/oxAuth#999]
Even when both applications supports OP/RP logouts user can get OP error page with error:
```
[xdi.oxauth.session.ws.rs.EndSessionRestWebServiceImpl] (EndSessionRestWebServiceImpl.java:165) - Session was removed successfully but redirect to post_logout_redirect_uri fails since AS failed to validate it against clients associated with session (which was just removed).
```

This happens when user user log into RP1 but trigger logout from RP2.

How to handle this with better user experience? I offer to redirect user to RP where user log in instead of showing OP error page. Do you have other alternatives of this idea?

**Comment by yuriyz** [comment]
User is logged in to RP1 and to RP2. If RP2 triggered logout then RP1 is notified about logout via `frontchannel_logout_uri`. RP1 is notified and should not call end_session endpoint and thus not get that error. @yurem Do I understand use case correctly?

**Comment by yurem** [comment]
Company has many RP1... RPn. As result user might click logout in RP where he is no logged in this session. This will show him OP error page.

**Comment by yuriyz** [comment]
I'm not sure I understand use case, if RP is not logged in then how user can click logout? I assume this use case should be handled by RP. Logout should be called only when RP is logged in. Otherwise yes, we can redirect to  `post_logout_redirect_uri` without validation if it's white listed. It would definitely look better for end-user.

**Comment by yurem** [comment]
There are crazy users like I which has bookmark like this `https://<server>/identity/logout`. Of course I'm using this for testing but what if user is using browser session restore and just forget about old tab? After opening it again and clicking Logout he will get this OP error.

**Comment by yurem** [comment]
My idea is to redirect him to last logged in RP if he not logged in current one.

**Comment by yuriyz** [comment]
Moved to 4.0 since it's related to https://github.com/GluuFederation/oxAuth/issues/1153 and we have to handle it in the same way.

**Comment by yuriyz** [comment]
Fixed in 4.0 and doc is added.
